tests/multi_ptr/multi_ptr_convert_assignment_ops.h

```diff
@@ -0,0 +1,199 @@
+      bool res = false;
+      {
+        sycl::buffer<bool> res_buf(&res, sycl::range<1>(1));
+        sycl::buffer<T> val_buffer(&value, sycl::range<1>(1));
+        queue.submit([&](sycl::handler &cgh) {
+          auto res_acc =
+              res_buf.template get_access<sycl::access_mode::write>(cgh);
+          auto acc_for_mptr =
+              val_buffer.template get_access<sycl::access_mode::read>(cgh);
+
+          if constexpr (src_space ==
+                        sycl::access::address_space::global_space) {
+            auto val_acc =
+                val_buffer.template get_access<sycl::access_mode::read>(cgh);
+            cgh.single_task([=] {
+              const src_multi_ptr_t mptr_in(acc_for_mptr);
+              dst_multi_ptr_t mptr_out;
+
+              mptr_out = std::move(mptr_in);
+
+              // Check that second mptr has the same value as first mptr
+              res_acc[0] = *(mptr_out.get_raw()) == acc_for_mptr[0];
+            });
+          } else {
+            sycl::local_accessor<T> local_acc(r, cgh);
+            cgh.parallel_for(
+                sycl::nd_range<1>(r, r), [=](sycl::nd_item<1> item) {
+                  if constexpr (src_space ==
+                                sycl::access::address_space::local_space) {
+                    auto ref = local_acc[0];
+                    value_operations::assign(ref, expected_val);
+
+                    const src_multi_ptr_t mptr_in(acc_for_mptr);
+                    dst_multi_ptr_t mptr_out;
+
+                    mptr_out = std::move(mptr_in);
+
+                    // Check that second mptr has the same value as first mptr
+                    res_acc[0] = *(mptr_out.get_raw()) == acc_for_mptr[0];
+                  } else {
+                    T private_val =
+                        user_def_types::get_init_value_helper<T>(expected_val);
+
+                    const src_multi_ptr_t mptr_in = sycl::address_space_cast<
+                        sycl::access::address_space::generic_space, decorated,
+                        T>(&private_val);
+                    dst_multi_ptr_t mptr_out;
+
+                    mptr_out = std::move(mptr_in);
+
+                    // Check that second mptr has the same value as first mptr
+                    res_acc[0] = *(mptr_out.get_raw()) == acc_for_mptr[0];
+                  }
+                });
+          }
+        });
+      }
+      CHECK(res);
+    }
+  }
+};
+
+template <typename T>
+class check_multi_ptr_convert_assign_for_type {
+ public:
+  void operator()(const std::string &type_name) {
+    const auto address_spaces = multi_ptr_common::get_address_spaces();
+    const auto is_decorated = multi_ptr_common::get_decorated();
+
+    // Run test with address_space and decorated for source multi_ptr type and
+    // with decorated for destination multi_ptr type
+    for_all_combinations<run_convert_assignment_operators_tests, T>(
+        address_spaces, is_decorated, is_decorated, type_name);
```

Comment:
Do I understand it correctly that source and destination `decorated` value is the same? I think the spec allows them to be different. It's not clearly stated for assign operator, but it's the case for implicit conversion.

Reply:
> Do I understand it correctly that source and destination decorated value is the same?

Not exactly, we have two same sets values of `sycl::access::decorated` field and we iterate over all combinations of them:
[here](https://github.com/KhronosGroup/SYCL-CTS/pull/349/files/e0d02a8749143b3fc292b1ebb264dfab2637b406#diff-aaaedf36f568a5a0b02e7455e96e48b9f6c1f06093a504cca944e7edb623965dR26-R29) we have `src_decorated` and `dst_decorated`, we will have the following combinations of them:
- `src_decorated` is `sycl::access::decorated::yes`, `dst_decorated` is `sycl::access::decorated::yes`
- `src_decorated` is `sycl::access::decorated::yes`, `dst_decorated` is `sycl::access::decorated::no`
- `src_decorated` is `sycl::access::decorated::no`, `dst_decorated` is `sycl::access::decorated::yes`
- `src_decorated` is `sycl::access::decorated::no`, `dst_decorated` is `sycl::access::decorated::no`

Therefore we will have all possible combinations of `sycl::access::decorated` enum's fields

Reply:
I see. Just the way this template is invoked confuses me. It looks like `src_decorated` and `dst_decorated` are assigned the same value. The thing I missed is that it's not a single value, but the entire set of possible values of `sycl::access::decorated`.